### PR TITLE
Alerts page bug fixes

### PR DESCRIPTION
**Pull request by Pixelguymm**
Requires beatmaps-io/beatsaver-common-mp#9

### CODE REVIEW

File: src/jsMain/kotlin/io/beatmaps/user/alerts/main.kt [hunk 1]
```diff
@@ -54,6 +54,7 @@ class AlertsPage : RComponent<AlertsPageProps, AlertsPageState>() {
 
             setState {
                 alertStats = data
+                hiddenAlerts = listOf()
```
Comment (Top-Cat) [inline src/jsMain/kotlin/io/beatmaps/user/alerts/main.kt:57]: Literally can't tell the difference, there shouldn't be one

File: src/jsMain/kotlin/io/beatmaps/user/alerts/alert.kt [hunk 2]
```diff
@@ -78,11 +80,13 @@ class AlertElement : RComponent<AlertProps, AlertState>() {
                 if (props.hidden == true) {
                     height = "0px"
                     opacity = "0"
+                    margin = "-1px 5px" // -1 pixel to account for the border
```
Comment (Top-Cat) [inline src/jsMain/kotlin/io/beatmaps/user/alerts/alert.kt:83]: If you're trying to account for the border.... why not change the border?

File: src/jsMain/kotlin/io/beatmaps/user/alerts/main.kt (hunk 1 above)
Comment (Pixelguymm) [reply src/jsMain/kotlin/io/beatmaps/user/alerts/main.kt:57]: When logging out, logging in again and going to your alerts page, the alerts don't disappear when you click the eye. Once you change the a filter or go to your read alerts, they do.

File: src/jsMain/kotlin/io/beatmaps/user/alerts/alert.kt (hunk 2 above)
Comment (Pixelguymm) [reply src/jsMain/kotlin/io/beatmaps/user/alerts/alert.kt:83]: As far as I could tell, the border width couldn't be animated with `transition`.

Comment (Top-Cat) [reply src/jsMain/kotlin/io/beatmaps/user/alerts/alert.kt:83]: `transition: opacity .2s, height .2s linear .2s, border-width .2s linear .2s`
works just fine if you actually define that property

File: src/jsMain/kotlin/io/beatmaps/user/alerts/main.kt (hunk 1 above)
Comment (Top-Cat) [reply src/jsMain/kotlin/io/beatmaps/user/alerts/main.kt:57]: Ah, would probably prefer this be handled on line 192
`hiddenAlerts = (hiddenAlerts ?: emptyListOf()).plus(id)`

**Pull request merged**